### PR TITLE
Update everything to .NET 8

### PR DESCRIPTION
**Pull request by xperiandri**

### CODE REVIEW

**Comment by xperiandri** [comment]
@Tarmil, @granicz  looks like this error is expected
https://sl.bing.net/ghTsdC7KdeS
> Source Branch: The error message mentions Action was triggered by pull_request: using SHA from head of source branch. If the action is running on a pull request from a forked repository, the token will have read-only permissions. This is a security measure on GitHub’s part.

**Comment by Tarmil** [comment]
So should we just not try to push test results from pull requests?

**Comment by xperiandri** [comment]
@Tarmil try the latest fix in this branch

**Comment by xperiandri** [comment]
It may help

**Comment by xperiandri** [comment]
@Tarmil approve again

File: src/Bolero.Server/Bolero.Server.fsproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project Sdk="Microsoft.NET.Sdk.Web">
   <PropertyGroup>
-    <TargetFrameworks>net6.0;net7.0;net8.0</TargetFrameworks>
```
Comment (Tarmil) [inline src/Bolero.Server/Bolero.Server.fsproj:4]: Unfortunately this is a breaking change.

There was a change in ASP.NET Core 7, namely a method added in an interface, which we had to take into account here: https://github.com/fsbolero/Bolero/blob/1eb9ee3f3ce11968f6052f1a82d355f4a3d1ef33/src/Bolero.Server/Remoting/Endpoints.fs#L153-L155

So if we remove net7.0 as a target, then .NET 7 applications will fall back on the net6.0 version of Bolero.Server, which is missing this part.

**Review by Tarmil — CHANGES_REQUESTED** [review]

File: src/Bolero.Server/Bolero.Server.fsproj (hunk 1 above)
Comment (xperiandri) [reply src/Bolero.Server/Bolero.Server.fsproj:4]: OK, I’ll fix

**Comment by xperiandri** [comment]
@Tarmil fixed

**Comment by xperiandri** [comment]
@Tarmil @granicz approve again, please

**Comment by xperiandri** [comment]
@Tarmil ping

**Comment by xperiandri** [comment]
@Tarmil checks passed, your comment addressed

**Comment by Tarmil** [comment]
Thanks!

**Pull request merged**